src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -53,23 +48,34 @@ public async Task CreateCdkDeployment(OrchestratorSession session, CloudApplicat
             {
                 // Create a new temporary CDK project for a new deployment
                 _interactiveService.LogMessageLine($"Generating a {recommendation.Recipe.Name} CDK Project");
-                cdkProjectPath = CreateCdkProjectForDeployment(recommendation, session);
+                cdkProjectPath = CreateCdkProject(recommendation, session);
             }
-            
+
             // Write required configuration in appsettings.json
             var appSettingsBody = _appSettingsBuilder.Build(cloudApplication, recommendation, session);
             var appSettingsFilePath = Path.Combine(cdkProjectPath, "appsettings.json");
-            using (var appSettingsFile = new StreamWriter(appSettingsFilePath))
+            await using var appSettingsFile = new StreamWriter(appSettingsFilePath);
+            await appSettingsFile.WriteAsync(appSettingsBody);
+
+            return cdkProjectPath;
+        }
+
+        public async Task DeployCdkProject(OrchestratorSession session, string cdkProjectPath, Recommendation recommendation)
+        {
+            var recipeInfo = $"{recommendation.Recipe.Id}_{recommendation.Recipe.Version}";
+            var environmentVariables = new Dictionary<string, string>
             {
-                await appSettingsFile.WriteAsync(appSettingsBody);
-            }
+                { EnvironmentVariableKeys.AWS_EXECUTION_ENV, recipeInfo }
+            };
 
             _interactiveService.LogMessageLine("Starting deployment of CDK Project");
 
             // Ensure region is bootstrapped
             await _commandLineWrapper.Run($"npx cdk bootstrap aws://{session.AWSAccountId}/{session.AWSRegion}",
                 needAwsCredentials: true);
 
+            var appSettingsFilePath = Path.Combine(cdkProjectPath, "appsettings.json");
```

Comment:
If this is a saved deployment project we should still be writing the `appsettings.json` file to `~/.aws-dotnet-deploy/<random-folder>`. We don't want to be writing a temporary file into the user's source control folder.

Reply:
This should be a separate task, this change is required in order to split the `CreateCdkDeployment` into two parts which are `ConfigureCdkProject` and `DeployCdkProject`.